### PR TITLE
Scorched zombie audit

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
Content "Scorched zombie audit"

#### Purpose of change
Our scorched zombies were weird living smoke grenades that for some reason were more armored than zombie cops (in all but bash). I really liked their concept, but hated the execution, and when working with them on my previous zombie PRs it's become apparent they've needed some TLC for a longer while.

#### Describe the solution
<details>
<summary>Changes to vanilla scorched and child scorched zombies</summary>

- Removed their natural armor weakpoints, because they're no longer naturally armored
- Slowed them down
- Upped the health for the adults, slightly lowered it for the kids
- Removed the burn care proficiency because there's no way cutting apart a horribly burnt zombie corpse is gonna teach you how to be a first responder
- Lowered their armor a bunch
- Removed their weird smoke grenade death spells
- Gave them low but not 0 bleed rates
- Lowered the adult's grab strength
- Slightly changed the adult's description
- Renamed the child to be more descriptive in what it actually is
- Normalized their melee to be slightly worse than their not-burnt counterparts

</details>
<details>
<summary>Scorched brute</summary>

- Based on the old fiend, replaced all occurences of old fiend ID with this
- Slightly weaker but more armored burnt version of a zombie brute

</details>
<details>
<summary>Fiend</summary>

- Complete overhaul to get it from a weird spiky DOOM-flavored brute into a burnt body horror gorilla-hulk
- Significantly stronger (in theory does more damage than the hulk, but is slower and more flimsy)
- Made the hulks (excluding the nemesis, as I don't want to touch it in case I break something horribly, unless @RenechCDDA or @PatrikLundell give me the blessing to do so as they're the people I've seen touch the nemesis last) burn into the fiends

</details>
<details>
<summary>Lasher</summary>

- Stat-wise basically the same as a default scorched zombie but slightly slower and worse at melee
- Doesn't grab and doesn't scratch, but utilizes its arms as whips to bash you from 3 tiles away
- Produced from batwing and grabber zombie lines

</details>
<details>
<summary>Dogs</summary>

- Pretty much nothing interesting to say, just scorched equivalents of zombie dogs, with a hulking horror equivalent present
- Baseline gets created from zombie dogs and their mutants, mutant coyotes, dogs, mutant dogs, wolves, zombie wolves
- Hulking horror equivalent gets created from hulking horrors or dire wolves

</details>

**MISC**
- Switched "pierces" for "rakes" in the stretch attack's definition, since piercing seems like it'd do more than the attack actually does, and most monsters would probably slash instead of stabbing directly. The exception is the naiad, which already has unique strings and is unaffected.
- Removed some unused lines from the batwing zombie family that I missed in my previous ``copy-from`` PRs. No gameplay effects from this.

#### Describe alternatives you've considered
The scorched variation of the hulking horror has a placeholder name because I couldn't think of anything better that was descriptive and didn't sound stupid. I initially wanted to go with Hellhound but that sounds cringe as hell

#### Additional context
- This, in general, is a massive nerf for the scorched zombie line. That is okay, as they are supposed to be more damaged variations of other zombies, not something specifically evolved. In future PRs I'd like to add a new zombie line that these guys would specifically evolve into, covered head to toe in scar tissue and *actually* armored. But that's for the future, because if I did it here the PR would be unreviewable.
- Lasher's special attack is defined separately instead of inside the monster definition as I intend to reuse it for the callous version of the lasher in a future PR, so I'd rather define the special in that file already than move file contents around later.

### CODE REVIEW

File: data/json/monsters/zed_burned.json [hunk 1]
```diff
@@ -2,82 +2,93 @@
...
+    "extend": { "flags": [ "REVIVES_HEALTHY", "NO_NECRO" ] }
   },
   {
-    "id": "mon_zombie_fiend",
+    "id": "mon_zombie_scorched_brute",
     "type": "MONSTER",
-    "name": { "str": "fiend" },
-    "description": "A charred zombie covered with bony plates, spikes, and protrusions.  Moves stiffly, but swiftly.",
+    "name": { "str": "scorched zombie brute" },
+    "description": "An oversized charred human form.  The remains of its burnt and blackened musculature clearly visible between the cracks of the shell formed from its melted skin.",
```
Comment (harakka) [inline data/json/monsters/zed_burned.json:22]:
```suggestion
    "description": "An oversized charred human form.  The remains of its burnt and blackened musculature are clearly visible between the cracks of the shell formed from its melted skin.",
```

File: data/json/monsters/zed_burned.json [hunk 2]
```diff
@@ -2,82 +2,93 @@
...
+    "description": "A heavily burned zombie that still reeks of charred meat.  Its now leathery flesh has mended into patches, giving it a more armored appearance.  This one appears to have metal fragments sticking out of its body.",
     "copy-from": "mon_zombie_scorched",
     "harvest": "zombie_rust"
   },
   {
     "id": "mon_zombie_fiend_rust",
     "type": "MONSTER",
-    "description": "A charred zombie covered with bony plates, spikes, and protrusions.  Moves stiffly, but swiftly.  This one appears to have metal fragments sticking out of its body.",
-    "copy-from": "mon_zombie_fiend",
+    "description": "An oversized charred human form.  The remains of its burnt and blackened musculature clearly visible between the cracks of the shell formed from its melted skin.  This one appears to have metal fragments sticking out of its body.",
```
Comment (harakka) [inline data/json/monsters/zed_burned.json:90]:
```suggestion
    "description": "An oversized charred human form.  The remains of its burnt and blackened musculature are clearly visible between the cracks of the shell formed from its melted skin.  This one appears to have metal fragments sticking out of its body.",
```

**Review by harakka — COMMENTED** [review]
I don't like the "mended into patches" thing but I don't know how to phrase it better.

**Comment by Maleclypse** [comment]
Conflicts

**Pull request merged**